pkg/kubelet/kubelet.go

```diff
@@ -1267,13 +1268,25 @@ func (kl *Kubelet) SyncPods(pods []api.BoundPod) error {
 		pc := podContainer{podFullName, uid, containerName}
 		if _, ok := desiredContainers[pc]; !ok {
 			glog.V(1).Infof("Killing unwanted container %+v", pc)
-			err = kl.killContainer(container)
+			err = kl.killContainer(dockerContainers[ix])
 			if err != nil {
 				glog.Errorf("Error killing container %+v: %v", pc, err)
+			} else {
+				killed = append(killed, dockerContainers[ix].ID)
 			}
 		}
 	}
 
+	running, err := dockertools.GetRunningContainers(kl.dockerClient, killed)
+	if err != nil {
+		glog.Errorf("Failed to poll container state: %v", err)
+		return err
+	}
+	if len(running) != 0 {
+		glog.Infof("Some containers still running: %v", running)
+		return nil
+	}
+
 	// Remove any orphaned pods.
 	err = kl.cleanupOrphanedPods(pods)
```

Comment:
could we do something like:

```
podsToKeepVolumes = pods
for each container in killed {
    extract pod UID from container name
    add UID to podsToKeepVolumes
}
kl.cleanupOrphanedPods(podsToKeepVolumes)
```

That way truly orphaned volumes will be cleaned up each sync loop, but pods that have some container alive are left for later